libs/test-utils/src/console.ts

```diff
@@ -127,28 +130,29 @@ export function suppressingConsoleOutput<T>(
   const warnStats = capturedCallCounts.get('warn') as { count: number };
   const errorStats = capturedCallCounts.get('error') as { count: number };
 
-  jest.spyOn(console, 'log').mockImplementation(() => {
+  const { spyOn } = await getTestRunner();
+  const logSpy = spyOn(console, 'log').mockImplementation(() => {
     logStats.count += 1;
   });
-  jest.spyOn(console, 'warn').mockImplementation(() => {
+  const warnSpy = spyOn(console, 'warn').mockImplementation(() => {
     warnStats.count += 1;
   });
-  jest.spyOn(console, 'error').mockImplementation(() => {
+  const errorSpy = spyOn(console, 'error').mockImplementation(() => {
     errorStats.count += 1;
   });
 
   function cleanup() {
-    jest.spyOn(console, 'log').mockRestore();
-    jest.spyOn(console, 'warn').mockRestore();
-    jest.spyOn(console, 'error').mockRestore();
+    logSpy.mockRestore();
+    warnSpy.mockRestore();
+    errorSpy.mockRestore();
```

Comment:
Seeming behavior difference: `jest` uses the original spy with this second `spyOn` call, then restores it. `vitest` seems to replace the first spy with a second spy, then immediately restores back to the first spy.